packages/ag-charts-community/src/chart/series/cartesian/cartesianSeries.ts

```diff
@@ -40,6 +40,10 @@ export interface CartesianSeriesNodeDatum extends SeriesNodeDatum {
     readonly yKey?: string;
     readonly xValue?: any;
     readonly yValue?: any;
+    // Caps can appear on bar, line and scatter series. The length is determined
+    // by the diameter of the marker (line, scatter), width of the bar (vertical
+    // bars), or height of the bar (horizontal bars)
+    readonly capLength?: number;
```

Comment:
`capLength` is a bit of misnomer, because this is actually the multiplier for `cap.lengthRatio`. Some other names to consider:

1. `capLengthRatioMultiplier`
2. `capMaxLength`

The first option is preferred because it's more specific and less likely to be misinterpreted. It is however much more verbose.

The second option is acceptable, but can be misleading because there are no checks in place that prevent the user from inputting a `cap.length` > `capMaxLength`, and such a check is probably undesirable (e.g. someone might want error bar caps that are wider than the markers on a line/scatter series).

Please let me know your thoughts.

Reply:
This has been changed to an object `capDefaults?: { lengthRatio: { defaultValue: number; multiplier: number } }`. Just using 1 number is insufficient because the defaults or line/scatter depends on the marker size, where bar series depend on the ratio size and rect width. So it's necessary to include also a "default ratio" to address this discrepancy.